### PR TITLE
test: copy search tests and use catalog

### PR DESCRIPTION
**Pull request by mkenigs**
In search.bats:
- Copy all tests relevant to the catalog and modify to use mock responses
- Run all irrelevant tests with the catalog feature enabled but empty mock responses
- Skip tests that require searching by version as that is not yet implemented

Tests that weren't copied include:
- Tests for number of results
- Tests for match strategy, which is no longer supported
- Tests for loading the lockfile from an environment
- Tests for specific packages like python310Packages, rubyPackages - those should be moved serverside. A test for printing pkg-path was added to make up for any coverage of printing `.` separated pkg-paths.
- A test for partial matching on pkg-path - that should be moved serverside.

### CODE REVIEW

File: cli/tests/catalog_responses/search/ello_--all.json
Comment (zmitchell) [inline cli/tests/catalog_responses/search/ello_--all.json:1]: Is this supposed to be "hello_--all" or something like that? It looks like the script mangled the name.

**Review by zmitchell — COMMENTED** [review]
It looks like some tests have `catalog: ` in the name, and others that use the catalog don't. Is that just an oversight?

**Comment by zmitchell** [comment]
I guess for the sake of time we're not moving any of these to unit tests?

File: cli/tests/catalog_responses/search/ello_--all.json
Comment (mkenigs) [reply cli/tests/catalog_responses/search/ello_--all.json:1]: No I wanted something with more than 10 results but still pretty small. `ello` was 25. Can change if that's weird

**Comment by mkenigs** [comment]
> I guess for the sake of time we're not moving any of these to unit tests?

That and I think there's more benefit to porting things that have to do something slow like a build, which we can avoid in a unit test. `search` is basically just pretty printing JSON

**Review by zmitchell — APPROVED** [review]

**Pull request merged**